tests/Collection/ArrayCollectionArrayAccessTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Arrays\Tests\Collection;
+
+use ArrayIterator;
+use PHPUnit\Framework\TestCase;
+use Yiisoft\Arrays\Collection\ArrayCollection;
+use Yiisoft\Arrays\Collection\ArrayCollectionIsImmutableException;
+use Yiisoft\Arrays\Collection\Modifier\RemoveKeys;
+use Yiisoft\Arrays\Collection\Modifier\UnsetValue;
+
+final class ArrayCollectionArrayAccessTest extends TestCase
+{
+    public function testIterator()
+    {
+        $collection = new ArrayCollection(
+            ['a' => 1, 'b' => 2, 'c' => 3],
+            new RemoveKeys()
+        );
+        $iterator = $collection->getIterator();
+
+        $this->assertInstanceOf(ArrayIterator::class, $iterator);
```

Comment:
Isn't it declared as return type already?

Reply:
Yes. I'm copy tests from ArrayAccessTraitTest.